File: app/src/main/java/com/test/xyz/demo/domain/repository/api/ProjectListRepository.java
package com.test.xyz.demo.domain.repository.api;

import com.test.xyz.demo.domain.model.GitHubRepo;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Path;
import rx.Observable;

public interface ProjectListRepository {
    String HTTPS_API_GITHUB_URL = "https://api.github.com/";

    @GET("users/{user}/repos")
    Observable<List<GitHubRepo>> getProjectList(@Path("user") String user);

    @GET("/repos/{user}/{reponame}")
    Observable<GitHubRepo> getProjectDetails(@Path("user") String user, @Path("reponame") String repoName);
}

File: app/src/main/kotlin/com/test/xyz/demo/domain/repository/api/ProjectListRepository.kt
package com.test.xyz.demo.domain.repository.api

import com.test.xyz.demo.domain.model.github.GitHubRepo

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

interface ProjectListRepository {

    @GET("users/{user}/repos")
    fun getProjectList(@Path("user") user: String): Observable<List<GitHubRepo>>

    @GET("/repos/{user}/{reponame}")
    fun getProjectDetails(@Path("user") user: String, @Path("reponame") repoName: String): Observable<GitHubRepo>

    companion object {
        val HTTPS_API_GITHUB_URL = "https://api.github.com/"
    }
}
